src/sagemaker/session.py

```diff
@@ -645,8 +645,12 @@ def logs_for_job(self, job_name, wait=False, poll=5):  # noqa: C901 - suppress c
 
             if state == LogState.JOB_COMPLETE:
                 state = LogState.COMPLETE
+            elif seconds_since_last_describe_job_call < 30:
+                seconds_since_last_describe_job_call += max(poll, 1)
```

Comment:
why not use actual seconds (from time.time()) instead of this pseudo-time counter?